designs/virtualization/4.x/create-vm-dropdowns/create-vm-dropdowns.md

```diff
@@ -0,0 +1,111 @@
+
+
+![Create VM - General-default](img/1-0.jpg)
+Landing on the General step, Default mode.
+Section titles added, and fields reorgenized and renamed.
+'General' section:
+- 'Name'
+- 'Description'
+'Resources & configuration' section'
+- 'Operating system'
+- 'VM configuration' (previously Workload profile)
+- 'Flavor (CPU and Memory)' (previously only 'flavor')
+'Operating system image source' section
+- 'Boot operating system via' (previously 'source')
+
+## General step
+
+![Create VM - General](img/1-1.jpg)
+On selecting an Operating system, the VM configuration and Flavor (CPU and Memory) fields will be enabled and selected on default
+
+![Create VM - General - VM configuration](img/1-1-1.jpg)
+VM Configuration options
+Server is the default
+
+![Create VM - General - flavor](img/1-1-2.jpg)
+Flavor options
+Medium is the default
+
+![Create VM - General](img/1-2-1.jpg)
+'Boot operating system via' field options
+
+![Create VM - General](img/1-2.jpg)
+Selecting an option form the 'Boot operating system via' field will expose below all the fields required to connect the VM to an OS image source.
+For example: on selecting the 'Existing PVC' option, additional fields will appear below, required to select a PVC available on the cluster.
+The disk name should be added here as well to establish a better link between this step and the Storage step.
+An infobox will notify the user that a disk will be added to to the disks list on the Storage step.
+
+![Create VM - OS disk on list](img/1-3.jpg)
+Within the storage step, the disk added on the general step will appear as the first item on the list.
+This disk cannot be deleted but can be modified.
+
+![Create VM - OS disk modal](img/1-3-1.jpg)
+Editing that disk will display a warning notification within the edit modal.
+
+## Netwroking step
+
+![Create VM - Networking -list](img/2-0.jpg)
+NIC list.
+In case no PXE boot the OS image selected on the General step, a default NIC will be added.
+This NIC  exposes the VM to the cluster.
+
+![Create VM - Networking -Default](img/2-1.jpg)
+On editing this default NIC, a warning message will remind the user that modifying this NIC might break that 'expoing to cluster' configuration.
+
+![Create VM - Networking -Add](img/2-2.jpg)
+Add NIC modal.
+The Advanced drawer is closed by default, as seen in the previous image.
+All fields had moved into the advanced drawer, except the Name and Network fields.
+Network field does not require description for each option since these options are created and managed by the user.
+
+![Create VM - Networking -Model](img/2-3.jpg)
+'Model' dropdown options
+
+![Create VM - Networking -Binding](img/2-4.jpg)
+'Binding method' dropdown options (previously 'Type')
+
+## Storage
+
+![Create VM - Storage -list](img/3-0.jpg)
+Disks list.
+In case no disk type OS image source selected in the General step, a default disk will be created in order to make the VM operational.
+
+![Create VM - Storage -Default disk](img/3-0-0.jpg)
```

Comment:
You can only delete this though if there is another disk right? So should we be more clear about that?

Reply:
This is a problem, cause now we have to say it to all the "last" disks.
How about we allow deleting the last disk but warn the user on an empty state that at least one must be created? Right now, except for the list description (which might be too far above) which I added, the user can't guess why she cannot delete it unless she edits it within a modal.

Reply:
ya not against this but I think we need to pull in the technical side - see my above comment :)

Reply:
We are not allowing delete action for the source disk currently. Is it useful to have such feature?

Reply:
It always feels a bit strange to lock it and not tell the user why. I think they should have the ability but be informed what the consequences would be.

Reply:
I suppose yes, the user should have a choice to do that. But it has to be sync with the boot source of the vm.

Reply:
So, we're making the default disk deletable, right?
Are we allowing to delete the disk containing the OS (defined in the OS step) as well?

Reply:
I thought this was what we were discussing. In case the user deletes the OS disk, we should probably let the user know he has to select other source / disk. Also we should probably detect the source according to the next boot order automatically.

Anyway, removal should probably not disable the disk tab and the source should not be a hard required field in the general tab. Relevant BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1840958

Reply:
@suomiy @matthewcarleton 
Revisiting this convo after we agreed to add the "OS source" step.

Do we agree that:
1. Because the OS source disk has been created externally, and because some fields within it are locked, we should lock this disk and mention it is created on the "OS source" step?
2. Because we mention to the user that the VM needs at least one disk available (regardless of the OS source disk), can we allow the user to delete the default disk?
3. When the user chooses to delete the last disk, we can pop up a confirmation modal, mentioning (yellow?) that the user will need to create a new disk instead.

WDYT guys?

Reply:
I suppose we could go this way, but what are we going to do about the validation of the OS source step?
When you delete the disk the step stops being valid. But the next steps will be filled already.

Reply:
We shouldn't allow an action to undo a previous step. It should be locked IMO and then we determine how to teach the user what the need to do.

Reply:
We all agree then.
I'll allow the deletion of default disks, with a notification that at least one disk is needed.
The OS source disk, in the case created, is not. deletable.

Reply:
When you change a boot order you can cause the validation to fail easily.

IMO we could do what we do in the VM Import wizard. We could allow everything and then show an error alert to highlight what validation was broken in OS image selection step. WDYT?